app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -101,23 +104,25 @@ object FirebaseDB: DB {
         }
     }
 
-    override suspend fun storeUserProfilePicture(image: Bitmap, userId: String, metadata: PhotoMetadata): PhotoMetadata? {
+    override suspend fun storeUserProfilePicture(image: Bitmap, userId: String, metadata: ProfilePhotoMetadata): ProfilePhotoMetadata? {
         if (userId.isBlank())
             throw java.lang.IllegalArgumentException("The userId cannot be a blank string")
 
-        metadata.pictureId = "${userId}_profile_picture"
+        val newMetadata = metadata.copy(pictureId = "${userId}_profile_picture")
 
-        return storePicture(image, USER_PROFILE_PICTURE_METADATA, metadata,
+        return storePicture(image, USER_PROFILE_PICTURE_METADATA, newMetadata,
             databaseReference.child(USERS_PATH).child(userId),
             storageReference.child(USERS_PATH).child(userId).child(
-                USER_PROFILE_PICTURE_METADATA))
+                USER_PROFILE_PICTURE_METADATA)) as ProfilePhotoMetadata
     }
 
-    override suspend fun addImage(image: Bitmap, userId: Int, metadata: PhotoMetadata): PhotoMetadata? {
+    override suspend fun addImage(image : Bitmap, photoMetadata: TrashPhotoMetadata): TrashPhotoMetadata? {
 
-        return storePicture(image, null, metadata,
-            databaseReference.child("pictures").child(userId.toString()),
-            storageReference.child("userId").child(userId.toString()))
+        var newPhotoMetadata = photoMetadata.copy(pictureId = null)
+
+        return storePicture(image, USER_POSTS, newPhotoMetadata,
+            databaseReference.child(USERS_PATH).child(newPhotoMetadata.takenBy!!).child(USER_POSTS),
+            storageReference.child(USERS_PATH).child(newPhotoMetadata.takenBy!!).child(USER_POSTS)) as TrashPhotoMetadata
     }

```

Comment:
This `addImage` method was not intended to be modified to serve as a `addPost` method. Since this method does not work as expected, I suggest you to look at `storeUserProfilePicture` and to make a similar method `storeNewPost` (or named differently if you find a better name of course), since those two methods do almost the same